pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -522,16 +550,19 @@ private InstancePartitions getInstancePartitions(TableConfig tableConfig,
             TableConfigUtils.hasPreConfiguredInstancePartitions(tableConfig, instancePartitionsType);
         if (hasPreConfiguredInstancePartitions) {
           String referenceInstancePartitionsName = tableConfig.getInstancePartitionsMap().get(instancePartitionsType);
+          InstancePartitions existingInstancePartitions = InstancePartitionsUtils.fetchInstancePartitions(
```

Comment:
Here we need to read the table's instance partitions name instead of the reference instance partitions name